modules/View.js

```diff
@@ -198,13 +198,24 @@
 
     /**
      * If detached, will replace the element passed in with this view's element and activate the view.
-     * @param $el [jQuery element] the element to attach to. This element will be replaced will this view
+     * @param $el {jQuery element} the element to attach to. This element will be replaced will this view
+     * @param [options] {Object} optional options
+     * @param   [options.replaceMethod] {Fucntion} if given, this view will invoke replaceMethod function
+     *                                             in order to attach the view's DOM to the parent instead of calling $el.replaceWith
+     * @param   [options.discardInjectionSite=false] {Booleon} if set to true, the injection site is not saved.
      * @method attach
      */
-    attach: function($el) {
+    attach: function($el, options) {
```

Comment:
unfortunate $el arg needs to be kept for backwards compatability.  Could we look at the object to determine whether it is providing a replaceMethod or not?  Then have a single polymorphic argument?